pkg/agent/baker.go

```diff
@@ -585,7 +585,7 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 			return ""
 		},
 		"IsKrustlet": func() bool {
-			return profile.KubeletType == "Krustlet"
+			return strings.EqualFold(profile.KubeletType, "Krustlet")
```

Comment:
one more nit: please make this a const similar to `datamodel.TempDiskContainerDataDir` above